src/app/qgsrulebasedlabelingwidget.cpp

```diff
@@ -27,6 +28,27 @@
 #include <QClipboard>
 #include <QMessageBox>
 
+
+static QList<QgsExpressionContextScope*> _globalProjectAtlasMapLayerScopes( QgsMapCanvas* mapCanvas, const QgsMapLayer* layer )
+{
+  QList<QgsExpressionContextScope*> scopes;
+  scopes << QgsExpressionContextUtils::globalScope()
+  << QgsExpressionContextUtils::projectScope( QgsProject::instance() )
+  << QgsExpressionContextUtils::atlasScope( nullptr );
+  if ( mapCanvas )
+  {
+    scopes << QgsExpressionContextUtils::mapSettingsScope( mapCanvas->mapSettings() )
+    << new QgsExpressionContextScope( mapCanvas->expressionContextScope() );
+  }
+  else
+  {
+    scopes << QgsExpressionContextUtils::mapSettingsScope( QgsMapSettings() );
+  }
+  scopes << QgsExpressionContextUtils::layerScope( layer );
```

Comment:
I was thinking the same - I have just copied the code from the original places without any changes to logic. There are some other places in the code that check whether layer is not null and add layer scope only we have a valid layer. I am happy to add not-null check for either project and layer as that makes more sense than just adding dummy scopes...